pkg/synchronization/controller.go

```diff
@@ -497,52 +496,58 @@ func (c *Controller) buildNotifications(spec creator_v1.ServiceSpec) (*orch_meta
 	}, nil
 }
 
-func (c *Controller) setupDockerSecret(logger *zap.Logger, namespace *core_v1.Namespace) (bool /* retriable */, *core_v1.Secret, error) {
+func (c *Controller) setupDockerSecret(logger *zap.Logger, namespaceName string) (bool /* retriable */, error) {
 	dockerSecret, err := c.MainClient.CoreV1().Secrets(dockerSecretNamespace).Get(dockerSecretName, meta_v1.GetOptions{})
 	if err != nil {
-		return true, nil, errors.Wrapf(err, "failed retrieving secret %q in namespace %q", dockerSecretName, dockerSecretNamespace)
+		return true, errors.Wrapf(err, "failed retrieving Secret %q in Namespace %q", dockerSecretName, dockerSecretNamespace)
 	}
 
-	secret := dockerSecret.DeepCopy()
-
 	// Check this secret is actually a docker config
-	if secret.Type != core_v1.SecretTypeDockerConfigJson {
-		return false, nil, errors.Errorf("secret %q in namespace %q is not a docker config secret", dockerSecretName, dockerSecretNamespace)
+	if dockerSecret.Type != core_v1.SecretTypeDockerConfigJson {
+		return false, errors.Errorf("Secret %q in Namespace %q is not a Docker config secret", dockerSecretName, dockerSecretNamespace)
 	}
 
-	// Set the namespace in the spec to the one we intend to copy to
-	secret.Namespace = namespace.Name
-
-	// Reset the UID, owner references and resourceVersion
-	secret.UID = ""
-	secret.ObjectMeta.OwnerReferences = nil
-	secret.ResourceVersion = ""
+	secret := &core_v1.Secret{
```

Comment:
Instead of mutilating an existing object construct a new one, copying only the needed things.